internal/provisioner/objects/service/service_test.go

```diff
@@ -177,14 +177,19 @@ func TestDesiredEnvoyService(t *testing.T) {
 	checkServiceHasPortName(t, svc, "https")
 	checkServiceHasPortProtocol(t, svc, corev1.ProtocolTCP)
 
+	cntr.Spec.NetworkPublishing.Envoy.Type = model.ClusterIPServicePublishingType
+	svc = DesiredEnvoyService(cntr)
+	checkServiceHasExternalTrafficPolicy(t, svc, corev1.ServiceExternalTrafficPolicyTypeLocal)
```

Comment:
In this case there should be _no_ external traffic policy, since it's invalid to set it for a ClusterIP service. Please modify the assertion to ensure that the field is not set.